3-async-await/solution.js

```diff
@@ -15,21 +18,60 @@ Example:
  price:1
 }
 
-7. both methods include some conditions to fail, at the end you should console.log() the errors, the format is up to you
+7. both methods include some conditions to fail,
+    at the end you should console.log() the errors,
+    the format is up to you
 8. add any needed adjustment to solution() function
-9. as extra challenge: add Promise.race() and Promise.any(), and try to get the idea of what happens
+9. as extra challenge: add Promise.race() and Promise.any(),
+    and try to get the idea of what happens
 */
 
 function solution() {
-    // YOUR SOLUTION GOES HERE
+  // imports
+  const getPrice = require("./prices.js");
+  const getProduct = require("./products.js");
+  //extract last 2 digit of Date.now string
+  const createdId = Date.now().toString().slice(-2);
 
-    // You generate your id value here
+  const arrayOfPromises = [getProduct(createdId), getPrice(createdId)];
+  const promiseAll = Promise.all(arrayOfPromises);
+  const promiseAllSettled = Promise.allSettled(arrayOfPromises);
 
-    // You use Promise.all() here
+  // call to async function to resolve each case
+  resolveAndPrintAsynchronously(promiseAll);
+  resolveAndPrintAsynchronously(promiseAllSettled);
 
-    // You use Promise.allSettled() here
+  async function resolveAndPrintAsynchronously(promise) {
+    try {
+      const [product, price] = await promise;
+      // console.log(product);
+      // console.log(price);
 
-    // Log the results, or errors, here
+      //Promise.all returns the same array with resolved Promises
+      if (typeof product === "string") {
```

Comment:
I think you can use destructuring to avoid checking the type of the result, that way you already know the type of the value